lib/internal/source_map/source_map_cache.js

```diff
@@ -105,12 +105,10 @@ function extractSourceMapURLMagicComment(content) {
 function maybeCacheSourceMap(filename, content, cjsModuleInstance, isGeneratedSource, sourceURL, sourceMapURL) {
   const sourceMapsEnabled = getSourceMapsEnabled();
   if (!(process.env.NODE_V8_COVERAGE || sourceMapsEnabled)) return;
-  try {
-    const { normalizeReferrerURL } = require('internal/modules/helpers');
-    filename = normalizeReferrerURL(filename);
-  } catch (err) {
+  const { normalizeReferrerURL } = require('internal/modules/helpers');
+  filename = normalizeReferrerURL(filename);
+  if (filename === undefined) {
```

Comment:
This function could really use a JSDoc, if you know what it should be. What is `filename`? Is it a path string? Because `normalizeReferrerURL` returns an URL string; shouldn’t it be one or the other?

This feature should support `data` URLs and `--experimental-network-imports`, so path strings should probably be normalized to URL strings so that non-`file://` URLs work.

Reply:
> This function could really use a JSDoc, if you know what it should be. 

I don't know fully what this does but filename is supposed to be always a string because it's used as keys to `SafeMap`s below.

> shouldn’t it be one or the other?

The current behavior is that "it doesn't matter as long as this string identifies a module that contains the sourceURL that V8 gives back to us". I think there can be some cache misses caused by this but that's probably out of scope of this PR, because this PR just tries to preserve the existing mapping behavior in `normalizeReferrerURL`. 

> This feature should support data URLs and --experimental-network-imports, so path strings should probably be normalized to URL strings so that non-file:// URLs work.

Supporting those for source maps seems to be out of scope for this PR?

Reply:
The filename here can either be an absolute path string (for CJS) or a URL string (for ESM) and is normalized to a URL string by `normalizeReferrerURL`. I can add JSDoc for source map related functions in a follow-up PR.